plutus-core/plutus-core/src/PlutusCore/Flat.hs

```diff
@@ -315,23 +309,24 @@ instance Flat a => Flat (Token a)
 instance Flat Keyword
 instance Flat Special
 
-deriving newtype instance Flat Index
+instance Flat Index where
+    -- encode from word to natural
```

Comment:
Needs a comment explaining why we're doing this! We spent some time talking about why we have to do it, let's write that down for future us!